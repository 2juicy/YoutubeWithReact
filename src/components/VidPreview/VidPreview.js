import React from "react";
import "./VidPreview.scss";

export default function VidPreview() {
  return <div>kjawhdjkahdjwahdaw placeholder</div>;
}
