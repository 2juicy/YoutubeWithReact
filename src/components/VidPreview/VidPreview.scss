@import "../../styles/shared.scss";

.vid-preview {
  display: grid;
  grid: 118px auto / 210px;
}

.vid-info {
  grid-row: 2 / 3;
  grid-column: 1 / 3;
  color: black;

  .vid-preview-metadata-container {
    padding-top: 5px;
    font-size: 13px;
    color: #6e6e6e;

    .channel-title {
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
    }
  }

  .show-max-two-lines {
    overflow: hidden;
    line-height: 1.4em;
    max-height: 2.8em;
  }

  .semi-bold {
    font-weight: bold;
  }
}

.image-container {
  position: relative;
  grid-row: 1 / 2;
  grid-column: 1 / 2;

  .time-label {
    position: absolute;
    background-color: black;
    bottom: 0;
    right: 0;
    opacity: 0.8s;
    border-radius: 2px;
    font-weight: 500;
    color: white;
    margin: 4px;
    padding: 2px 4px;
    line-height: 12px;
  }
}
