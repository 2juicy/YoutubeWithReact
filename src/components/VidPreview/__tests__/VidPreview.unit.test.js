import React from "react";
import { shallow } from "enzyme";
import VidPreview from "../VidPreview";

describe("VidPreview", () => {
  test("renders", () => {
    const wrapper = shallow(<VidPreview />);
    expect(wrapper).toMatchSnapshot();
  });
});
