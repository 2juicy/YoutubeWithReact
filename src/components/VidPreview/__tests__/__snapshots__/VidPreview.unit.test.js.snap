// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VidPreview renders 1`] = `
<div
  className="vid-preview"
>
  <div
    className="image-container"
  >
    <Image
      as="img"
      src="http://via.placeholder.com/210x118"
      ui={true}
    />
    <div
      className="time-label"
    >
      <span>
        04:20
      </span>
    </div>
  </div>
  <div
    className="vid-info"
  >
    <div
      className="vid-title show-max-two-lines"
    >
      Video Title
    </div>
    <div
      className="vid-preview-metadata-container"
    >
      <div
        className="channel-title"
      >
        Channel Title
      </div>
      <p>
        24M views Â· 22 days ago
      </p>
    </div>
  </div>
</div>
`;
